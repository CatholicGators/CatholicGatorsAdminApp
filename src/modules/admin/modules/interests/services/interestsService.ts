import 'firebase/firestore'
import { NewOptionReq, NewSectionReq } from '../redux/actions/interestActions'
import FirestoreAdapter from '../../../../../database/firestoreAdapter'

type SectionDoc = {
    id: string
    text: string
    position: number
    options: string[]
}

export type Section = {
    id: string
    text: string
    position: number
    options: Option[]
}

export type Option = {
    id: any
    text: string
}

export default class InterestsService {
    public static readonly OPTIONS: string = 'options'
    public static readonly SECTIONS: string = 'sections'

    constructor(private adapter: FirestoreAdapter) { }

    async getInterests(): Promise<Section[]> {
        const sectionDocs = await this.adapter.getAll<SectionDoc>(InterestsService.SECTIONS)
        const optionDocs = await this.adapter.getAll<Option>(InterestsService.OPTIONS)
        const sections = sectionDocs.map(section => ({
            ...section,
            options: section.options
                .map(
                    optionId =>
                        optionDocs.filter(option => option.id === optionId)[0]
                )
                .filter(option => option !== undefined)
        }))
        return sections.sort((a, b) => a.position - b.position)
    }

    addOption(sectionId: string, optionReq: NewOptionReq): Promise<Option> {
        return this.adapter.runTransaction(async transaction => {
            const optionDoc = await transaction.get(
                this.adapter.getNewDocReference(InterestsService.OPTIONS)
            )
            const sectionDoc = await transaction.get(
                this.adapter.getDocReference(InterestsService.SECTIONS, sectionId)
            )

            const options = sectionDoc.data().options
            options.push(optionDoc.id)
            transaction.update(sectionDoc.ref, {
                options
            })
            transaction.set(optionDoc.ref, optionReq)

            return {
                id: optionDoc.id,
                ...optionReq
            }
        })
    }

    updateOptionText(optionId: string, text: string): Promise<Option> {
        return this.adapter.update<Option>(InterestsService.OPTIONS, optionId, {
            text
        })
    }

    addSection(sectionReq: NewSectionReq): Promise<Section> {
        return this.adapter.add<Section>(InterestsService.SECTIONS, sectionReq)
    }
}
