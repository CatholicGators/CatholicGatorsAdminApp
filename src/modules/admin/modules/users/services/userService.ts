import FirestoreAdapter, { Doc } from '../../../../../database/firestoreAdapter'

export interface UserData {
    name: String
    email: String
    photoURL: String
    isApproved: boolean
    isAdmin: boolean
}
export interface User extends Doc, UserData { }

export default class UserService {
    public static readonly USERS: string = 'users'

    constructor(private adapter: FirestoreAdapter) { }

    getUser(id: string): Promise<User> {
        return this.adapter.get<User>(UserService.USERS, id)
    }

    getAllUsers(): Promise<User[]> {
        return this.adapter.getAll<User>(UserService.USERS)
    }

    updateApproval(id: string, isApproved: boolean): Promise<User> {
        return this.adapter.update<User>(UserService.USERS, {
            id,
            changes: { isApproved }
        })
    }

    // updateUserAdminStatus(id: string, isAdmin: boolean): Observable<User> {
    //     return this.docRefObservableToUserObservable(
    //         this.db.updateDoc(USER_COLLECTION, id, {
    //             isAdmin
    //         })
    //     )
    // }

    // updateUsersAdminStatus(
    //     ids: string[],
    //     isAdmin: boolean
    // ): Observable<User[]> {
    //     const updates = ids.map(id => ({
    //         id,
    //         isAdmin
    //     }))

    //     return this.docRefsObservableToUsersObservable(
    //         this.db.updateDocs(USER_COLLECTION, updates)
    //     )
    // }

    // deleteUser(id: string): Observable<void> {
    //     return this.db.deleteDoc(USER_COLLECTION, id)
    // }

    // deleteUsers(ids: string[]): Observable<void> {
    //     return this.db.deleteDocs(USER_COLLECTION, ids)
    // }
}
