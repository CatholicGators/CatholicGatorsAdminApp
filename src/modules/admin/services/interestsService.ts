import 'firebase/firestore'
import { NewOptionReq, NewSectionReq } from '../redux/actions/interestActions';

export type SectionDoc = {
    id: string
    text: string
    position: number
    options: string[]
}

export type Section = {
    id: string
    text: string
    position: number
    options: Option[]
}

export type Option = {
    id: any
    text: string
}

interface Doc {
    id: string
}

export default class InterestsService {
    public static readonly OPTIONS: string = 'options'
    public static readonly SECTIONS: string = 'sections'

    constructor(private db: firebase.firestore.Firestore) {}

    async getInterests() : Promise<Section[]> {
        const [sectionDocs, optionDocs] = await Promise.all([
            this.getAllAndFlatten<SectionDoc>(InterestsService.SECTIONS),
            this.getAllAndFlatten<Option>(InterestsService.OPTIONS)
        ]);
        const sections = sectionDocs.map(section => ({
            ...section,
            options: section.options.map(optionId => optionDocs.filter(option => option.id === optionId)[0]).filter(option => option !== undefined)
        }));
        return sections.sort((a, b) => a.position - b.position);
    }

    addOption(sectionId: string, optionReq: NewOptionReq) : Promise<Option> {
        return this.db.runTransaction(async transaction => {
            const optionDoc = await transaction.get(this.db.collection(InterestsService.OPTIONS).doc())
            const sectionDoc = await transaction.get(this.db.collection(InterestsService.SECTIONS).doc(sectionId))

            const options = sectionDoc.data().options
            options.push(optionDoc.id)
            transaction.update(sectionDoc.ref, {
                options
            })
            transaction.set(optionDoc.ref, optionReq)

            return {
                id: optionDoc.id,
                ...optionReq
            }
        })
    }

    updateOptionText(optionId: string, newText: string) : Promise<Option> {
        return this.db.runTransaction(async transaction => {
            const optionDoc = await transaction.get(this.db.collection(InterestsService.OPTIONS).doc(optionId))
            const updatedOption = {
                ...optionDoc.data(),
                text: newText
            }

            transaction.set(optionDoc.ref, updatedOption)

            return {
                id: optionId,
                ...updatedOption,
            }
        })
    }

    async addSection(sectionReq: NewSectionReq) : Promise<Section> {
        const sectionRef = await this.db.collection(InterestsService.SECTIONS).add(sectionReq);
        return ({
            id: sectionRef.id,
            ...sectionReq
        });
    }

    private async getAllAndFlatten<T extends Doc>(collection: string) : Promise<T[]> {
        const snapshot = await this.db.collection(collection).get();
        return snapshot.docs.map(doc => ({
            id: doc.id,
            ...doc.data()
        } as T));
    }
}
