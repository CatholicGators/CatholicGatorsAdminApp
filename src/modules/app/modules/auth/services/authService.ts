import { auth } from "firebase/app"
import "firebase/auth"
import { Observable, from } from "rxjs"

import UserService, {
    User,
} from "../../../../admin/modules/users/services/userService"
import { switchMap } from "rxjs/operators"

type Auth = auth.Auth
type FirebaseUser = firebase.User

export default class AuthService {
    private user$: Observable<User>

    constructor(private auth: Auth, private userService: UserService) {
        this.user$ = Observable.create((observer) =>
            this.auth.onAuthStateChanged(
                (user) => observer.next(user),
                (err) => observer.error(err)
            )
        ).pipe(
            switchMap((x: FirebaseUser) => from(this.storeUserInfoOnLogin(x)))
        )
    }

    googleSignIn(): Promise<void> {
        return this.auth.signInWithRedirect(new auth.GoogleAuthProvider())
    }

    signOut(): Promise<void> {
        return this.auth.signOut()
    }

    listenForUser(): Observable<User> {
        return this.user$
    }

    private storeUserInfoOnLogin(user: FirebaseUser): Promise<User> {
        if (user) {
            return this.userService.getOrInitUser(user)
        } else {
            return Promise.resolve(null)
        }
    }
}
