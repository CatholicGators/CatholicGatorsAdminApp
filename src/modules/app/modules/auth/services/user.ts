import Document from "../../../../../models/document";

export default interface User extends Document {
    name: String;
    email: String;
    photoUrl: String;
    isApproved: boolean;
    isAdmin: boolean;
}
