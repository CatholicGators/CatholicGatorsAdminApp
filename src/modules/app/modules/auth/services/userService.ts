import { auth } from "firebase/app";
import "firebase/auth";
import { Observable, merge, from, zip } from "rxjs";
import { mergeMap, map, partition } from "rxjs/operators";
import Firestore from "../../../../../database/firestore";
import User from "./user";

const USER_COLLECTION = "users";
const ADMIN_EMAIL = "sandy@catholicgators.org"; // TODO: we might want to migrate this to some form of config

export default class UserService {
    private db: Firestore;
    private auth: auth.Auth;
    private firebaseUser$: Observable<firebase.User>;
    private user$: Observable<User>;

    constructor(firestore: Firestore) {
        this.db = firestore;
        this.auth = firestore.getAuth();

        this.firebaseUser$ = Observable.create((observer) =>
            this.auth.onAuthStateChanged(
                (user) => observer.next(user),
                (err) => observer.error(err)
            )
        );
        this.user$ = this.storeUserInfoOnLogin();
    }

    googleSignIn(): Observable<void> {
        return from(
            this.auth.signInWithRedirect(new auth.GoogleAuthProvider())
        );
    }

    signOut(): Observable<void> {
        return from(this.auth.signOut());
    }

    listenForUser(): Observable<User> {
        return this.user$;
    }

    addUser(user: User): Observable<User> {
        return this.docRefObservableToUserObservable(
            this.db.addDoc(USER_COLLECTION, user)
        );
    }

    addUsers(users: User[]): Observable<User[]> {
        return this.docRefsObservableToUsersObservable(
            this.db.upsertDocs(USER_COLLECTION, users)
        );
    }

    getUser(id: string): Observable<User> {
        return this.db.getDoc(USER_COLLECTION, id) as Observable<User>;
    }

    getAllUsers(): Observable<User[]> {
        return this.db.getCollection(USER_COLLECTION) as Observable<User[]>;
    }

    updateUser(id: string, update: Object): Observable<User> {
        return this.docRefObservableToUserObservable(
            this.db.updateDoc(USER_COLLECTION, id, update)
        );
    }

    updateUsers(users: User[]): Observable<User[]> {
        return this.docRefsObservableToUsersObservable(
            this.db.updateDocs(USER_COLLECTION, users)
        );
    }

    updateUserApproval(id: string, isApproved: boolean): Observable<User> {
        return this.docRefObservableToUserObservable(
            this.db.updateDoc(USER_COLLECTION, id, {
                isApproved,
            })
        );
    }

    updateUsersApproval(
        ids: string[],
        isApproved: boolean
    ): Observable<User[]> {
        const updates = ids.map((id) => ({
            id,
            isApproved,
        }));

        return this.docRefsObservableToUsersObservable(
            this.db.updateDocs(USER_COLLECTION, updates)
        );
    }

    updateUserAdminStatus(id: string, isAdmin: boolean): Observable<User> {
        return this.docRefObservableToUserObservable(
            this.db.updateDoc(USER_COLLECTION, id, {
                isAdmin,
            })
        );
    }

    updateUsersAdminStatus(
        ids: string[],
        isAdmin: boolean
    ): Observable<User[]> {
        const updates = ids.map((id) => ({
            id,
            isAdmin,
        }));

        return this.docRefsObservableToUsersObservable(
            this.db.updateDocs(USER_COLLECTION, updates)
        );
    }

    deleteUser(id: string): Observable<void> {
        return this.db.deleteDoc(USER_COLLECTION, id);
    }

    deleteUsers(ids: string[]): Observable<void> {
        return this.db.deleteDocs(USER_COLLECTION, ids);
    }

    private storeUserInfoOnLogin(): Observable<User> {
        const attachExistence = (firebaseUser) =>
            this.db
                .doesExist(USER_COLLECTION, firebaseUser.uid)
                .pipe(map((exists) => ({ exists, firebaseUser })));

        const storeUser = ({ exists, firebaseUser }) => {
            const docId = firebaseUser.uid;
            const user: User = this.wrapFirebaseUser(firebaseUser);
            let observable;

            if (exists) {
                // If the user already exists, just update their profile pic
                observable = this.db.updateDoc(USER_COLLECTION, docId, {
                    photoUrl: user.photoUrl,
                });
            } else {
                observable = this.db.upsertDocById(
                    USER_COLLECTION,
                    docId,
                    user
                );
            }

            return observable.pipe(
                mergeMap((_) =>
                    this.db.getDoc(USER_COLLECTION, firebaseUser.uid)
                )
            ) as Observable<User>;
        };

        let [login$, logout$] = partition((x) => x !== null)(
            this.firebaseUser$
        );
        login$ = login$.pipe(mergeMap(attachExistence), mergeMap(storeUser));

        return merge(login$, logout$) as Observable<User>;
    }

    private wrapFirebaseUser(user: firebase.User): User {
        return {
            id: user.uid,
            name: user.displayName,
            email: user.email,
            photoUrl: user.photoURL,
            isApproved: user.email === ADMIN_EMAIL,
            isAdmin: user.email === ADMIN_EMAIL,
        };
    }

    private docRefObservableToUserObservable(
        docRef$: Observable<firebase.firestore.DocumentReference>
    ): Observable<User> {
        return docRef$.pipe(
            mergeMap((ref) => from(ref.get())),
            map(
                (snapshot) =>
                    ({
                        id: snapshot.id,
                        ...snapshot.data(),
                    } as User)
            )
        );
    }

    private docRefsObservableToUsersObservable(
        docRefs$: Observable<firebase.firestore.DocumentReference[]>
    ): Observable<User[]> {
        return docRefs$.pipe(
            mergeMap((refs) => {
                const observables = refs.map((x) => from(x.get()));
                return zip(...observables);
            }),
            map((snapshots) => {
                return snapshots.map(
                    (snapshot) =>
                        ({
                            id: snapshot.id,
                            ...snapshot.data(),
                        } as User)
                );
            })
        );
    }
}
