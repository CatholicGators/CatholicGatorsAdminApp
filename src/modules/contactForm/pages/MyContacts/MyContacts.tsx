import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import classNames from 'classnames';
import {
  Theme,
  createStyles,
  withStyles,
  Paper,
  Table,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
  Toolbar,
  Typography,
  Card,
  CardContent,
  CardActions,
  Button,
  CircularProgress,
  Select,
  MenuItem,
  OutlinedInput
} from '@material-ui/core'

import {
    getContacts,
    updateContactStatus
} from '../../../../redux/actions/contactForm/contactFormActions'

const styles = (theme: Theme) => createStyles({
    tableWrapper: {
        maxWidth: '1000px',
        margin: '50px auto'
    },
    cardWrapper: {
        display: 'flex',
        justifyContent: 'center',
        marginTop: '50px',
        paddingLeft: theme.spacing.unit,
        paddingRight: theme.spacing.unit
    },
    tableCard: {
        overflowX: 'auto'
    },
    tableLoadingContainer: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        minHeight: '300px',
    },
    select: {
        width: '100%',
        borderRadius: 4
    },
    outlinedInput: {
        paddingTop: theme.spacing.unit,
        paddingBottom: theme.spacing.unit
    },
    notCalled: {
        background: 'rgb(234, 153, 153, 0.5)'
    },
    needToCall: {
        background: 'rgb(255, 229, 153, 0.5)'
    },
    called: {
        background: 'rgb(182, 215, 168, 0.5)'
    }
})

type Props = {
    history: any,
    user: any,
    classes: any,
    contacts: any,
    getContacts: () => void,
    updateContactStatus: (contact, status) => void
}

export class MyContacts extends Component<Props> {
    componentDidMount() {
        this.props.getContacts()
    }

    goToHome() {
        this.props.history.push('/')
    }

    changeContactStatus(contact, status) {
        this.props.updateContactStatus(contact, status)
    }

    getClassFromStatus(status, classes) {
        if(!classes) {
            return null
        }

        switch(status) {
            case "Called":
                return classes.called
            case "Need to call again":
                return classes.needToCall
            default:
                return classes.notCalled
        }
    }

    render() {
        const { classes, contacts, user } =  this.props
    
        return user && user.isApproved ? (
            <div className={classes ? classes.tableWrapper : null}>
                <Paper className={classes ? classes.tableCard : null}>
                    <Toolbar>
                        <Typography variant="h6">
                            My Contacts
                        </Typography>
                    </Toolbar>
                    {contacts ?
                        <Table>
                            <TableHead>
                                <TableRow>
                                    <TableCell>Name</TableCell>
                                    <TableCell>Phone Number</TableCell>
                                    <TableCell>Email</TableCell>
                                    <TableCell>Call Status</TableCell>
                                </TableRow>
                            </TableHead>
                            <TableBody>
                            {contacts.map(contact => (
                                <TableRow hover key={contact.id}>
                                    <TableCell>{`${contact.firstName} ${contact.lastName}`}</TableCell>
                                    <TableCell>{contact.phoneNumber}</TableCell>
                                    <TableCell>{contact.email}</TableCell>
                                    <TableCell>
                                        <Select
                                            className={classNames(
                                                classes ? classes.select : null,
                                                this.getClassFromStatus(contact.status, classes)
                                            )}
                                            value={contact.status || "Not called"}
                                            onChange={ev => this.changeContactStatus(contact, ev.target.value)}
                                            input={
                                                <OutlinedInput
                                                    classes={{ input: classes ? classes.outlinedInput : null }}
                                                    labelWidth={0}
                                                />
                                            }
                                        >
                                            <MenuItem value={"Not called"}>Not called</MenuItem>
                                            <MenuItem value={"Need to call again"}>Need to call again</MenuItem>
                                            <MenuItem value={"Called"}>Called</MenuItem>
                                        </Select>
                                    </TableCell>
                                </TableRow>
                            ))}
                            </TableBody>
                        </Table>
                    :
                        <div id='loading-spinner' className={classes ? classes.tableLoadingContainer : null}>
                            <CircularProgress size="60px" />
                        </div>
                    }
                </Paper>
            </div>
        ) : (
            <div className={classes ? classes.cardWrapper : null}>
                <Card>
                    <CardContent>
                        <Typography gutterBottom variant="h5" component="h2">
                            You are not approved
                        </Typography>
                        <Typography component="p">
                            Please seek approval from an admin
                        </Typography>
                    </CardContent>
                    <CardActions>
                        <Button color="primary" onClick={() => this.goToHome()}>Ok</Button>
                    </CardActions>
                </Card>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    contacts: state.contactForm.contacts,
    user: state.auth.user
})

const mapDispatchToProps = dispatch => ({
    getContacts: () => dispatch(getContacts()),
    updateContactStatus: (contact, status) => dispatch(updateContactStatus(contact, status))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(MyContacts)))
