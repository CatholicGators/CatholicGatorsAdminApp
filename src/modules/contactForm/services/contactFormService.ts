import InterestsService, {
    Section
} from '../../admin/modules/interests/services/interestsService'
import FirestoreAdapter, { Doc } from '../../../database/firestoreAdapter'

export const ContactStatus = {
    NOT_CALLED: 0,
    NEED_TO_CALL_AGAIN: 2,
    CALLED: 4
}

interface ContactData {
    firstName: string
    lastName: string
    gender: string
    email: string
    phoneNumber: string
    graduationSemester: string
    graduationYear: string
    school: string
    permanentAddress: string
    city: string
    state: string
    zipCode: string
    housingComplex: string
    parentName: string
    parentPhone: string
    parentEmail: string
    interests: string[]
    status: number
    createdAt: Date
}
export interface NewContactReq extends ContactData { }
export interface Contact extends Doc, ContactData { }

export default class ContactFormService {
    public static readonly CONTACTS_COLLECTION: string = 'contacts'

    constructor(
        private adapter: FirestoreAdapter,
        private interestsService: InterestsService
    ) { }

    addContact(contact: NewContactReq): Promise<Contact> {
        return this.adapter.add<Contact>(ContactFormService.CONTACTS_COLLECTION, contact)
    }

    getInterests(): Promise<Section[]> {
        return this.interestsService.getInterests()
    }

    getAllContacts(): Promise<Contact[]> {
        return this.adapter.getAll<Contact>(ContactFormService.CONTACTS_COLLECTION)
    }

    updateContactStatus(contactId: string, status: number): Promise<Contact> {
        return this.adapter.update<Contact>(ContactFormService.CONTACTS_COLLECTION, {
            id: contactId,
            changes: {
                status
            }
        })
    }
}
