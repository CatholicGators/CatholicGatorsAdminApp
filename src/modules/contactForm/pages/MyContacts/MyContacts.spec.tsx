import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"

import { CircularProgress, Card, CardActions, Button } from "@material-ui/core"
import { MyContacts } from "./MyContacts"
import { ContactStatus } from "../../services/contactFormService"

describe("MyContacts", () => {
    let props, wrapper, instance

    beforeEach(() => {
        props = {
            user: {
                isApproved: true,
            },
            history: {
                push: jest.fn(),
            },
            getContacts: jest.fn(),
            updateContactStatus: jest.fn(),
        }
        wrapper = shallow(<MyContacts {...props} />)
        instance = wrapper.instance()
    })

    it("should match snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    describe("user is not approved", () => {
        beforeEach(() => {
            wrapper.setProps({ user: undefined })
        })

        it("should show the redirect card", () => {
            expect(wrapper.exists("#redirect-card")).toBe(true)
        })

        it("should send the user home when the button is pressed", () => {
            const card = wrapper.find(Card)
            const cardActions = card.find(CardActions)
            const button = cardActions.find(Button)

            button.simulate("click")

            expect(props.history.push).toHaveBeenCalledWith("/")
        })
    })

    it("should call getContacts() on startup", () => {
        expect(props.getContacts).toHaveBeenCalled()
    })

    it("shows a spinner when loading the contacts", () => {
        expect(wrapper.exists(CircularProgress)).toBe(true)
    })

    describe("getClassFromStatus(status, classes)", () => {
        let classes = {
            called: "called",
            needToCall: "needToCall",
            notCalled: "notCalled",
        }

        it("returns null if classes is null", () => {
            const result = instance.getClassFromStatus(
                ContactStatus.CALLED,
                null
            )

            expect(result).toBe(null)
        })

        it("returns null if classes is undefined", () => {
            const result = instance.getClassFromStatus(
                ContactStatus.CALLED,
                undefined
            )

            expect(result).toBe(null)
        })

        it("returns classes.notCalled by default", () => {
            const result = instance.getClassFromStatus(12345678, classes)

            expect(result).toBe(classes.notCalled)
        })

        it("returns classes.notCalled when the status is not called", () => {
            const result = instance.getClassFromStatus(
                ContactStatus.NOT_CALLED,
                classes
            )

            expect(result).toBe(classes.notCalled)
        })

        it("returns classes.called when the status is Called", () => {
            const result = instance.getClassFromStatus(
                ContactStatus.CALLED,
                classes
            )

            expect(result).toBe(classes.called)
        })

        it('returns classes.needToCall when the status is "Need to call again"', () => {
            const result = instance.getClassFromStatus(
                ContactStatus.NEED_TO_CALL_AGAIN,
                classes
            )

            expect(result).toBe(classes.needToCall)
        })
    })

    describe("changeContactStatus(contact, status)", () => {
        it("calls updateContactStatus(contact, status)", () => {
            const contact = {}
            const status = 0

            instance.changeContactStatus(contact, status)

            expect(props.updateContactStatus).toHaveBeenLastCalledWith(
                contact,
                status
            )
        })
    })
})
