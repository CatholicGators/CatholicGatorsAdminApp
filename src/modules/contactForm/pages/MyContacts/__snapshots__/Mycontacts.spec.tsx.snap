// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyContacts should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyContacts
    getContacts={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Card)>
        <WithStyles(CardContent)>
          <WithStyles(Typography)
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            You are not approved
          </WithStyles(Typography)>
          <WithStyles(Typography)
            component="p"
          >
            Please seek approval from an admin
          </WithStyles(Typography)>
        </WithStyles(CardContent)>
        <WithStyles(CardActions)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
          >
            Ok
          </WithStyles(Button)>
        </WithStyles(CardActions)>
      </WithStyles(Card)>,
      "className": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(CardContent)>
            <WithStyles(Typography)
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              You are not approved
            </WithStyles(Typography)>
            <WithStyles(Typography)
              component="p"
            >
              Please seek approval from an admin
            </WithStyles(Typography)>
          </WithStyles(CardContent)>,
          <WithStyles(CardActions)>
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              Ok
            </WithStyles(Button)>
          </WithStyles(CardActions)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                component="h2"
                gutterBottom={true}
                variant="h5"
              >
                You are not approved
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                component="p"
              >
                Please seek approval from an admin
              </WithStyles(Typography)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "You are not approved",
                "component": "h2",
                "gutterBottom": true,
                "variant": "h5",
              },
              "ref": null,
              "rendered": "You are not approved",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Please seek approval from an admin",
                "component": "p",
              },
              "ref": null,
              "rendered": "Please seek approval from an admin",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              Ok
            </WithStyles(Button)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Ok",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Ok",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Card)>
          <WithStyles(CardContent)>
            <WithStyles(Typography)
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              You are not approved
            </WithStyles(Typography)>
            <WithStyles(Typography)
              component="p"
            >
              Please seek approval from an admin
            </WithStyles(Typography)>
          </WithStyles(CardContent)>
          <WithStyles(CardActions)>
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              Ok
            </WithStyles(Button)>
          </WithStyles(CardActions)>
        </WithStyles(Card)>,
        "className": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(CardContent)>
              <WithStyles(Typography)
                component="h2"
                gutterBottom={true}
                variant="h5"
              >
                You are not approved
              </WithStyles(Typography)>
              <WithStyles(Typography)
                component="p"
              >
                Please seek approval from an admin
              </WithStyles(Typography)>
            </WithStyles(CardContent)>,
            <WithStyles(CardActions)>
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
              >
                Ok
              </WithStyles(Button)>
            </WithStyles(CardActions)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)
                  component="h2"
                  gutterBottom={true}
                  variant="h5"
                >
                  You are not approved
                </WithStyles(Typography)>,
                <WithStyles(Typography)
                  component="p"
                >
                  Please seek approval from an admin
                </WithStyles(Typography)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "You are not approved",
                  "component": "h2",
                  "gutterBottom": true,
                  "variant": "h5",
                },
                "ref": null,
                "rendered": "You are not approved",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Please seek approval from an admin",
                  "component": "p",
                },
                "ref": null,
                "rendered": "Please seek approval from an admin",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                color="primary"
                onClick={[Function]}
              >
                Ok
              </WithStyles(Button)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Ok",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Ok",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
